### PR TITLE
[Calling] Enhanced Support + Audio Only

### PR DESCRIPTION
**Pull request by ahammer**
## Purpose

* Enhanced Support + Audio Only

Enhanced support provides the OnUserReportedIssue event, and the support form integration. This will allow users to submit issues, and Contoso developers to capture them.

Audio Only provides a CallCompositeAvMode, This mode allows .normal and .audioOnly modes. In Audio only, local camera is disabled, as are remote video feeds. Screen sharing from remote will continue to function.

- Refactoring the `CallCompositeViewModelFactoryProtocol` into it's own file, to address SwiftLint file length limits

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [x] Documentation content changes
- [ ] Other... Please describe:

## Pull request checklist

This PR has considered:
- [x] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [x] Internationalization and Right to Left layouts
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [x] View Data Injection
- [x] Demo App UI/UX update

## How to Test

* Open...
* Click on...

## What to Check

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Screenshot or description before this change | Screenshot or description with this change |

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/DebugInfo.swift [hunk 1]
```diff
@@ -9,9 +9,20 @@ import Foundation
 public struct DebugInfo {
     /// The history of calls up to 30 days. Ordered ascending by call started date.
     public let callHistoryRecords: [CallHistoryRecord]
-
+    public let callingUIVersion: String
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/DebugInfo.swift:12]: Please add missing documentation

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift [hunk 2]
```diff
@@ -27,13 +30,23 @@ public struct LocalOptions {
                 setupScreenViewData: SetupScreenViewData? = nil,
                 cameraOn: Bool? = false,
                 microphoneOn: Bool? = false,
-                skipSetupScreen: Bool? = false
-    ) {
+                skipSetupScreen: Bool? = false,
+                avMode: CallCompositeAvMode = .normal) {
```
Comment (pavelprystinka) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift:34]: Please add documentation to Parameters. We apparently did already miss a few.

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift (hunk 2 above)
Comment (ahammer) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift:43]: This file has had it's documentation updated now

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallCompositeAvMode.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import Foundation
+
+public enum CallCompositeAvMode {
```
Comment (iaulakh) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallCompositeAvMode.swift:8]: misisng doc for public API

**Review by iaulakh — COMMENTED** [review]
Adam, t will be good to add few UI tests

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallCompositeAvMode.swift (hunk 3 above)
Comment (ahammer) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/CallCompositeAvMode.swift:8]: Thanks, will add now.

**Review by pavelprystinka — DISMISSED** [review]

**Review by pavelprystinka — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

**Review by pavelprystinka — APPROVED** [review]

**Pull request merged**